sonar-orchestrator/src/main/java/com/sonar/orchestrator/Orchestrator.java

```diff
@@ -245,9 +255,48 @@ private BuildResult executeBuildInternal(Build<?> build, boolean quietly, boolea
     return buildResult;
   }
 
+  private void setDefaultAdminToken(Build<?> build) {
+    if (build.getProperties().containsKey(SONAR_LOGIN_PROPERTY_NAME)) {
+      return;
+    }
+
+    if (build.arguments().stream().anyMatch(s -> s.contains(SONAR_LOGIN_PROPERTY_NAME))) {
+      return;
+    }
+
+    if (distribution.useDefaultAdminCredentialsForBuilds()) {
+      build.setProperty(SONAR_LOGIN_PROPERTY_NAME, getDefaultAdminToken());
+    }
+  }
+
+  public String getDefaultAdminToken() {
+    if (this.adminToken != null) {
+      return this.adminToken;
+    }
+    return generateDefaultAdminToken();
+  }
+
+  private String generateDefaultAdminToken() {
+    HttpCall httpCall = server.newHttpCall("api/user_tokens/generate")
+      .setParam("name", UUID.randomUUID().toString())
+      .setMethod(HttpMethod.POST)
+      .setAdminCredentials();
+    HttpResponse response = httpCall.execute();
+    if (response.isSuccessful()) {
+      this.adminToken = ofNullable(Json.parse(response.getBodyAsString()).asObject().get("token"))
+        .map(JsonValue::asString)
+        .orElseThrow(() -> new IllegalStateException("Could not extract admin token from response: " + response.getBodyAsString()));
+      return this.adminToken;
+    } else {
+      throw new IllegalStateException("Could not get token for admin: " + response.getBodyAsString());
+    }
+  }
+
   public BuildResult[] executeBuilds(Build<?>... builds) {
     requireNonNull(buildRunner, ORCHESTRATOR_IS_NOT_STARTED);
 
+    Arrays.stream(builds).forEach(this::setDefaultAdminToken);
```

Comment:
Do you think we should write a unit test to check that the default admin token is set for all builds?